src/material-experimental/mdc-snack-bar/snack-bar-container.ts

```diff
@@ -188,4 +205,36 @@ export class MatSnackBarContainer extends BasePortalOutlet
       throw Error('Attempting to attach snack bar content after content is already attached');
     }
   }
+
+  /**
+   * Starts a timeout to move the snack bar content to the live region so screen readers will
+   * announce it.
+   */
+  private _screenReaderAnnounce() {
+    if (this._nonLiveElementRef && this._liveElementRef) {
+      const nonLiveEle: HTMLElement = this._nonLiveElementRef.nativeElement;
+      const liveEle: HTMLElement = this._liveElementRef.nativeElement;
+
+      // If an element in the snack bar content is focused before being moved
+      // track it and restore focus after moving to the live region.
+      let previouslyFocusedEle: HTMLElement | null = null;
+      if (this._platform.isBrowser &&
```

Comment:
The `instanceof HTMLElement` check is because `document.activeElement` is type `Element` which doesn't implement `focus`. So I check to make sure the active element is of type `HTMLElement` and therefore focusable.

I found without including the `isBrowser` check I would get some view_engine test failures related to `document` not existing. For `mdc-snack-bar` with the switch back to using `setTimeout` the call to `_screenReaderAnnounce` is already nested in a `isBrowser` check, so I've removed the redundant one. For `snack-bar` I've kept the check because it isn't done elsewhere in the component. 

Here's an example test failure: https://app.circleci.com/pipelines/github/angular/components/13171/workflows/d5670bcf-b85d-4721-8372-86b96b5f872f/jobs/187667/parallel-runs/0/steps/0-107